android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt

```diff
@@ -140,14 +140,20 @@ constructor(
 
   suspend fun <R : Resource> addOrUpdate(addMandatoryTags: Boolean = true, resource: R) {
     return withContext(dispatcherProvider.io()) {
-      resource.updateLastUpdated()
-      try {
-        fhirEngine.get(resource.resourceType, resource.logicalId).run {
-          fhirEngine.update(updateFrom(resource))
-        }
-      } catch (resourceNotFoundException: ResourceNotFoundException) {
-        create(addMandatoryTags, resource)
+      if (resource.hasMeta() && resource.meta.hasLastUpdated()) {
```

Comment:
I think this check should always be true for all resources so I do not think it is necessary. Are the checks failing for any reason for particular resources?